src/browser/base/content/zen-styles/zen-tabs/vertical-tabs.css

```diff
@@ -472,12 +495,84 @@
 
       border-top-right-radius: var(--zen-border-radius);
       width: var(--zen-toolbox-max-width);
+      /* Closing transition */
+      transition: 
+        width var(--tabs-colapsing-time) var(--tabs-colapsing-timing),
+        border-top-color var(--tabs-colapsing-time) var(--tabs-colapsing-timing),
+        border-right-color var(--tabs-colapsing-time) var(--tabs-colapsing-timing),
+        /* Hide background-color slower, otherwise, background content can be seen */
+        background-color var(--tabs-fading-color-time) cubic-bezier( 0.99, 0, 0.83, 0.99 ) !important;
 
       & .tabbrowser-tab {
         transition: 0s !important;
       }
     }
 
+    /* Colapsed */
+    #navigator-toolbox:not(#navigator-toolbox:is(
+    #navigator-toolbox[zen-user-hover='true'][zen-has-hover],
+    #navigator-toolbox[zen-user-hover='true']:focus-within,
+    #navigator-toolbox[zen-user-hover='true'][movingtab],
+    #navigator-toolbox[zen-user-hover='true'][flash-popup],
+    #navigator-toolbox[zen-user-hover='true'][has-popup-menu],
+    #navigator-toolbox[zen-user-hover='true']:has(*[open='true']:not(tab):not(#zen-sidepanel-button)),
+    #navigator-toolbox[zen-expanded='true']:not([zen-user-hover='true']))) {
+      & #tabbrowser-tabs {
+        & .tabbrowser-tab {
+          width: unset !important;
+        }
+      }
+    }
+
+    /* Expanded */
+    #navigator-toolbox:is(
```

Comment:
I dont really understand this change, why not just add it at the top with the rest?

Reply:
I mostly didn't want change too much code.

Since there is a lot of possible layout combinations, when I tried change changing something like this, it lead to breaking one or all of them, which required more changes to fix and I ended up rewriting the whole file.

To not create a huge PR with the whole file rewritten, I tried to limit the changes to affect only the Expand on Hover layout and maintain the other ones unchanged. 

For these specific lines, those properties wore applied with !important some other place in the file, so the changes weren't being applied unless I matched the whole selector.

Reply:
But you right. I checked this again and moved to the top with the rest and seems to cause no problems

Should I remove it?